Fix/include search in discovery

apana03 opened the pull request:
## Summary

This PR updates the Discovery Tab in the search page to also include the query from the search bar while showing the feed.

## Changes

- **Screens/UI:** filters the trips returned by the getFeed function according to if it contains the search query or not, also updated the composable that is shown when no trips are found in the discover tab in order for it to be more appealing..

## Checklist

Ensure all / most of these are checked before the pull request is submitted.
- [x] This PR resolves #306.
- [x] Tests have been added for new functionality.
- [x] Screenshots or UI changes have been attached.
- [x] Documentation has been updated.

##  Testing


- **Manual Testing:**
    - [x] Tested on emulator / physical device.
- **Unit Tests**
    - [x] Added units tests for this


##  Related Issues/Tickets

Closes #306 

## Screenshots (if applicable)

https://github.com/user-attachments/assets/79abaa51-b09f-4781-91d6-de36c835c373

mirceagavrizi commented on app/src/main/java/com/android/voyageur/ui/search/DiscoverContent.kt:
```diff
@@ -142,30 +159,40 @@ fun TripCard(trip: Trip, userViewModel: UserViewModel) {
+            horizontalAlignment = Alignment.CenterHorizontally) {
+              Icon(
+                  imageVector = Icons.Outlined.ImageSearch,
+                  contentDescription = "No Trips Found",
```
Here I think you missed a hardcoded string, can you extract it in the xml file?

mirceagavrizi commented on app/src/main/java/com/android/voyageur/ui/search/SearchScreen.kt:
```diff
@@ -397,7 +397,7 @@ fun SearchScreen(
                 }
           } else {
-            DiscoverContent(tripsViewModel, userViewModel)
+            DiscoverContent(searchQuery.text, tripsViewModel, userViewModel)
```
Also, in this file at line 263, can you add the placeholder for the search bar for this screen to be "Search trips..."?

mirceagavrizi reviewed (commented):
Otherwise I checked it on my emulator and everything runs smoothly, great job adding this feature as it will significantly improve the UX. Let me know when you have extracted the strings and added the placeholder and I will happily approve!

apana03 commented:
> Otherwise I checked it on my emulator and everything runs smoothly, great job adding this feature as it will significantly improve the UX. Let me know when you have extracted the strings and added the placeholder and I will happily approve!

Addressed your comments. Can you take a look again?

mirceagavrizi reviewed (approved):
Thank you for quickly implementing my suggestions!
